src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs

```diff
@@ -50,17 +50,19 @@ protected virtual void ModelDirective()
         {
             // Verify we're on the right keyword and accept
             AssertDirective(ModelKeyword);
+            var startModelLocation = CurrentLocation;
             AcceptAndMoveNext();
+            var keywordLength = Span.GetContent().Value.Length;
```

Comment:
And move down just before it's used.  No reason to calculate at all unless adding an error.